### PR TITLE
Show popup

### PR DESCRIPTION
**Pull request by watasuke102**
## Context

Zen can't open client popup.

## Summary

Add view-chld, and handle view's `new_popup` signal, show popup.

## How to check behavior

1. start zen with some client (ex: Google Chrome)
2. click right button or menu button (three dot on the right-top of the view)

### CODE REVIEW

File: include/zen/view-child.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      double child_sy, double *toplevel_sx, double *toplevel_sy);
+};
+
+struct zn_view_child {
+  void *user_data;
+  struct wlr_surface *surface;  // nonnull
+
+  const struct zn_view_child_interface *impl;
+
+  struct zn_view *view;
```
[the remark this answers is not in the document]
Comment (watasuke102) [reply include/zen/view-child.h:19]: `xdg_popup` is destroyed at the same time of destroying `xdg_surface`, so it seems not to be required for now.

[the remark this answers is not in the document]
Comment (Aki-7) [reply include/zen/view-child.h:19]: What do you mean by "for now"? Are xdg_popups always destroyed when its parent xdg_surface (which is identical with this `zn_view`) is destroyed?

[the remark this answers is not in the document]
Comment (watasuke102) [reply include/zen/view-child.h:19]: I mean I'm not sure about the popup in xwayland.

[the remark this answers is not in the document]
Comment (Aki-7) [reply include/zen/view-child.h:19]: > Are xdg_popups always destroyed when their parent xdg_surface (which is identical with this zn_view) is destroyed?

How about this?

[the remark this answers is not in the document]
Comment (watasuke102) [reply include/zen/view-child.h:19]: Sorry it is more correct to say "`xdg_popup` is destroyed at the same time of destroying `xdg_toplevel`".

According to the wlroots (`types/xdg_shell/wlr_xdg_surface.c`),
- `wlr_xdg_popup_destroy` is called in `unmap_xdg_surface`
- It is called in `reset_xdg_surface`
- It is called in `destroy_xdg_toplevel` (in `wlr_xdg_toplevel.c`)

[the remark this answers is not in the document]
Comment (Aki-7) [reply include/zen/view-child.h:19]: Thanks for the clarification. That is exactly what I was wondering about. I would appreciate it if you could leave it in a brief comment.

```suggestion
  struct zn_view *view;   // nonnull, zn_view_child must be destroyed when its zn_view is destroyed. 
```

[the remark this answers is not in the document]
Comment (watasuke102) [reply include/zen/view-child.h:19]: OK, can I do that like following?
```c
/** automatically destroyed when its zn_view is destroyed */
struct zn_view_child {
  void *user_data;
  struct wlr_surface *surface;  // nonnull

  const struct zn_view_child_interface *impl;

  struct zn_view *view;  // nonnull
  // ...
```

[the remark this answers is not in the document]
Comment (Aki-7) [reply include/zen/view-child.h:19]: It programmer's responsibility to check that zn_view_child is automatically destroyed or not when its zn_view is destoyed.
In the case of xdg_popup, it's automatically destoyed though.

```c
/** Make sure that this is destroyed when its zn_view is destroyed */
struct zn_view_child {
  void *user_data;
  struct wlr_surface *surface;  // nonnull

  const struct zn_view_child_interface *impl;

  struct zn_view *view;  // nonnull
  // ...
```

**Review by Aki-7 — APPROVED** [review]
Super nice!

**Pull request merged**